staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/types.go

```diff
@@ -53,13 +53,23 @@ type APIServiceSpec struct {
 	// CABundle is a PEM encoded CA bundle which will be used to validate an API server's serving certificate.
 	CABundle []byte
 
-	// Priority controls the ordering of this API group in the overall discovery document that gets served.
-	// Client tools like `kubectl` use this ordering to derive preference, so this ordering mechanism is important.
-	// Values must be between 1 and 1000
-	// The primary sort is based on priority, ordered lowest number to highest (10 before 20).
+	// GroupPriority controls the ordering of the group with respect to other groups. The lowest of all GroupPriorities
+	// for the same group is used.  Must be greater than zero.
+	// The primary sort is based on GroupPriority, ordered lowest number to highest (10 before 20).
+	// The secondary sort is based on the VersionPriority, ordered lowest number to highest.
+	// The tertiary sort is based on the alphabetical comparison of the name of the object.  (v1.bar before v1.foo)
+	// We'd recommend something like: *.k8s.io (except extensions) at 1000, extensions at 2000
+	// PaaSes (OpenShift, Deis) are recommended to be in the 3000s
+	GroupPriority int32
```

Comment:
of those, only the separate resource representing the group is understandable to me

> order versions by kubernetes version scheme: {highest stable ... lowest stable, highest beta ... lowest beta, highest alpha ... lowest alpha, alphabetical for anything that doesn't follow the format}

we shouldn't hard-code semantic understanding of kube version names

> Order group by max(Priority) instead of min(Priority)

this doesn't help the problem of changing a version priority affecting group priority, it just reverses it

> Order group by median priority (probably most stable thing)

this is impossible to understand

>     Order group by (1+priority/100) and versions by priority %100 (a hack but at least a hack that doesn't make hash of the API)

this is impossible to understand